core/ecschema-editing/src/Editing/Constants.ts

```diff
@@ -0,0 +1,61 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Editing
+ */
+
+import { ConstantProps, DelayedPromiseWithProps, Phenomenon, SchemaItemKey, SchemaItemType, SchemaKey, SchemaMatchType } from "@bentley/ecschema-metadata";
+import { SchemaContextEditor, SchemaItemEditResults } from "./Editor";
+import { MutableConstant } from "./Mutable/MutableConstant";
+import { MutableSchema } from "./Mutable/MutableSchema";
+
+/**
+ * @alpha
+ * A class allowing you to create schema items of type Constant.
+ */
+export class Constants {
+  public constructor(protected _schemaEditor: SchemaContextEditor) { }
+
+  public async create(schemaKey: SchemaKey, name: string, phenomenon: SchemaItemKey, definition: string, displayLabel?: string, numerator?: number, denominator?: number): Promise<SchemaItemEditResults> {
+    const schema = (await this._schemaEditor.schemaContext.getCachedSchema(schemaKey, SchemaMatchType.Exact)) as MutableSchema;
+    if (schema === undefined) return { errorMessage: `Schema Key ${schemaKey.toString(true)} not found in context` };
+
+    const newConstant = (await schema.createConstant(name)) as MutableConstant;
+    if (newConstant === undefined) {
+      return { errorMessage: `Failed to create class ${name} in schema ${schemaKey.toString(true)}.` };
+    }
+
+    const newPhenomenon = (await this._schemaEditor.schemaContext.getSchemaItem(phenomenon)) as Phenomenon;
+    if (newPhenomenon === undefined || newPhenomenon.schemaItemType !== SchemaItemType.Phenomenon) {
+      return { errorMessage: `Unable to locate phenomenon ${phenomenon.name}` };
+    }
+    newConstant.setPhenomenon(new DelayedPromiseWithProps<SchemaItemKey, Phenomenon>(newPhenomenon.key, async () => newPhenomenon));
+    newConstant.setDefinition(definition);
+
+    if (numerator) { newConstant.setNumerator(numerator); }
```

Comment:
Our [coding guidelines](https://github.com/imodeljs/imodeljs/blob/master/docs/learning/guidelines/typescript-coding-guidelines.md#make-judicious-use-of-vertical-screen-space) specify the body of a conditional always goes on a separate line and omits braces if it consists only of a single line.